pkg/awsconfig/testing/firewallConfig.go

```diff
@@ -0,0 +1,410 @@
+		CidrBlock: aws.String(CidrBlock),
+		VpcId:     aws.String(*Vpc.Vpc.VpcId),
+	}
+	Subnet, err := c.ec2Client.CreateSubnet(Subnetinput)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			default:
+				fmt.Println(aerr.Error())
+			}
+		} else {
+			fmt.Println(err.Error())
+		}
+	}
+
+	return *Subnet
+}
+
+func (c Client) CreateRouteTable (Vpc ec2.CreateVpcOutput, Subnet ec2.CreateSubnetOutput)(ec2.CreateRouteTableOutput){
+	RouteTable1input := &ec2.CreateRouteTableInput{
+		VpcId: aws.String(*Vpc.Vpc.VpcId),
+	}
+
+	RT, err := c.ec2Client.CreateRouteTable(RouteTable1input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			default:
+				fmt.Println(aerr.Error())
+			}
+		} else {
+			fmt.Println(err.Error())
+		}
+	}
+	Associateinput := &ec2.AssociateRouteTableInput{
+		RouteTableId: aws.String(*RT.RouteTable.RouteTableId),
+		SubnetId:     aws.String(*Subnet.Subnet.SubnetId),
+	}
+	_ , err = c.ec2Client.AssociateRouteTable(Associateinput)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			default:
+				fmt.Println(aerr.Error())
+			}
+		} else {
+			fmt.Println(err.Error())
+		}
+	}
+	return *RT
+}
+
+func (c Client) CreateRouteTableIG (Vpc ec2.CreateVpcOutput, IG ec2.CreateInternetGatewayOutput)(ec2.CreateRouteTableOutput){
+	RouteTable1input := &ec2.CreateRouteTableInput{
+		VpcId: aws.String(*Vpc.Vpc.VpcId),
+	}
+
+	RT, err := c.ec2Client.CreateRouteTable(RouteTable1input)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			default:
+				fmt.Println(aerr.Error())
+			}
+		} else {
+			fmt.Println(err.Error())
+		}
+	}
+	Associateinput := &ec2.AssociateRouteTableInput{
+		RouteTableId: aws.String(*RT.RouteTable.RouteTableId),
+		GatewayId:     aws.String(*IG.InternetGateway.InternetGatewayId),
+	}
+	_ , err = c.ec2Client.AssociateRouteTable(Associateinput)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			default:
+				fmt.Println(aerr.Error())
+			}
+		} else {
+			fmt.Println(err.Error())
+		}
+	}
+	return *RT
+}
+func (c Client)CreateNatGateway( Subnet ec2.CreateSubnetOutput) (ec2.CreateNatGatewayOutput){
+	EIPinput := &ec2.AllocateAddressInput{
+		Domain: aws.String("vpc"),
+	}
+
+	EIPresult, err := c.ec2Client.AllocateAddress(EIPinput)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			default:
+				fmt.Println(aerr.Error())
+			}
+		} else {
+			fmt.Println(err.Error())
+		}
+	}
+	NGinput := &ec2.CreateNatGatewayInput{
+		AllocationId: aws.String(*EIPresult.AllocationId),
+		SubnetId:     aws.String(*Subnet.Subnet.SubnetId),
+	}
+
+	NGresult, err := c.ec2Client.CreateNatGateway(NGinput)
+	if err != nil {
+		if aerr, ok := err.(awserr.Error); ok {
+			switch aerr.Code() {
+			default:
+				fmt.Println(aerr.Error())
+			}
+		} else {
+			fmt.Println(err.Error())
+		}
+	}
+	time.Sleep(wait)
```

Comment:
Could you explain why there's a 2-minute wait here? If we're just waiting on AWS to bring up the NAT Gateway, maybe indicate that in the console output? When I ran into this delay initially, I assumed the script crashed.

Reply:
Any Sleeps we should refactor to retries with backoff

Reply:
Yeah that's to wait for the NAT Gateway to be ready. will add a log saying this.

Reply:
should we look into polling?